ui/packages/catalog-ui-search/src/main/webapp/component/upload/upload.view.js

```diff
@@ -41,16 +44,64 @@ define([
         },
         initialize: function(){
             this.listenTo(router, 'change', this.handleRoute);
-            this.handleRoute();
+            this.listenTo(uploadInstance, 'change:currentUpload', this.onBeforeShow);
         },
         handleRoute: function(){
-            if (router.toJSON().name === 'openUpload'){
-                this.$el.removeClass('is-hidden');
-            } else {
-                this.$el.addClass('is-hidden');
+            const routerJSON = router.toJSON();
+            if (routerJSON.name === 'openUpload'){
+                var uploadId = routerJSON.args[0];
+                var upload = user.get('user').get('preferences').get('uploads').get(uploadId);
+                if (!upload) {
+                    router.notFound();
+                } else {
+                    const queryForMetacards = new Query.Model({
+                        cql: cql.write({
+                            type: 'OR',
+                            filters: _.flatten(upload.get('uploads').filter(function(file){
+                                return file.id || file.get('children') !== undefined;
+                            }).map(function(file){
+                                if (file.get('children') !== undefined) {
+                                    return file.get('children').map((child) => ({
+                                        type: '=',
+                                        value: child,
+                                        property: '"id"'
+                                    }));
+                                } else {
+                                    return {
+                                        type: '=',
+                                        value: file.id,
+                                        property: '"id"'
+                                    };
+                                }
+                            }).concat({
```

Comment:
:question: I don't quite understand why this call is a `concat`. Am I correct in my analysis that this `Array.concat` call with a non-array element is equivalent to a `push`? If so, I would find it much more readable to change this to a `push` to clarify the intention of simply appending a single element.